libs9s/s9srpcclient.cpp

```diff
@@ -10304,15 +10314,62 @@ S9sRpcClient::composeBackupJob()
     if (!snapshotLocation.empty())
         jobData["snapshot_location"] = snapshotLocation;
 
-    if (options->hasCredentialIdOption())
-        jobData["credential_id"] = options->credentialId();
-
-    if (!s3bucket.empty())
-        jobData["s3_bucket"] = s3bucket;
-
     if (!s3region.empty())
         jobData["s3_region"] = s3region;
 
+    if (onlyCloud)
+    {
+        jobData["only_cloud"] = true;
+        // checks with current available functionality
+        if(cloudStorageProv.empty() || cloudStorageProv != AWS_CLOUD_PROVIDER)
+        {
+            PRINT_ERROR("Only supported S3 buckets. Use: "
+            "--cloud-provider=aws --s3-bucket=<my-bucket> options");
+            return S9sVariantMap();
+        }
+        else
+        {
+            // Using existing parameter names for create backup job when uploading backup
+            // upload_backup_to_cloud command expects following arguments
+            if (options->hasCredentialIdOption())
```

Comment:
```suggestion
if (!options->hasCredentialIdOption())
{
    PRINT_ERROR("Need option --credential-id=<id-num> with --only-cloud");
    return S9sVariantMap();
}

jobCloudData["cloud_storage_credentials_id"] = options->credentialId();
```
Try fail fast approach to reduce branches in code.